cmake/app/boilerplate.cmake

```diff
@@ -247,6 +247,17 @@ set(CMAKE_CXX_COMPILER_FORCED 1)
 include(${ZEPHYR_BASE}/cmake/version.cmake)
 include(${ZEPHYR_BASE}/cmake/host-tools.cmake)
 include(${ZEPHYR_BASE}/cmake/kconfig.cmake)
+# DTS should be run directly after kconfig because CONFIG_ variables
+# from kconfig and dts should be available at the same time. But
+# running DTS involves running the preprocessor, so we put it behind
+# toolchain. Meaning toolchain.cmake is the only component where
+# kconfig and dts variables aren't available at the same time.
+
+# dts.cmake requires a host compiler with C preprocessor.
+# Set it now; toolchain.cmake will set it again later with cross-compiler.
+find_program(CMAKE_C_COMPILER gcc)
```

Comment:
We can't do find_program(gcc) because we don't want a host C compiler as a dependency.

A host C compiler is a difficult dependency on Windows.

Reply:
OK, thanks for the feedback.
In that case, would the better solution be to leave dts.cmake after toolchain.cmake, and update dts.cmake to use different compiler options depending on the toolchain, for the dts preprocessor step?

Reply:
I believe so, but it must also be done in a way where dts.cmake is agnostic to the C compiler used. Any differences must be abstracted away before dts.cmake is executed, e.g. in toolchain.cmake.

So in dts.cmake it is OK to do:

`set(cmd ${CMAKE_C_COMPILER} ${PREPROCESS_FLAG})
`

But it is not OK to do:
```

if(TOOLCHAIN STREQUAL TI)
set(cmd ${CMAKE_C_COMPILER} -Preprocess)
else()
set(cmd ${CMAKE_C_COMPILER} -E)
endif()

```